src/software/ai/hl/stp/tactic/goalie_tactic.cpp

```diff
@@ -244,82 +245,60 @@ void GoalieTactic::calculateNextAction(ActionCoroutine::push_type &yield)
-
-        // compute block cone position, allowing 1 ROBOT_MAX_RADIUS_METERS extra on
-        // either side
-        Point goalie_pos = calcBlockCone(
-            field.friendlyGoalpostNeg(), field.friendlyGoalpostPos(), ball.position(),
-            block_cone_radius * block_cone_angle.toRadians());
-
-        // we want to restrict the block cone to the friendly crease, also potentially
-        // scaled by a defense_area_deflation_parameter
-        Rectangle deflated_defense_area = field.friendlyDefenseArea();
-        deflated_defense_area.expand(-defense_area_deflation);
-
-        // restrain the goalie in the deflated defense area, if the goalie cannot be
-        // restrained or if there is no proper intersection, then we safely default to
-        // center of the goal
-        auto clamped_goalie_pos =
-            restrainGoalieInRectangle(goalie_pos, deflated_defense_area);
-
-        // if the goalie could not be restrained in the deflated defense area,
-        // then the ball must be either on a really sharp angle to the net where
-        // its impossible to get a shot, or the ball is behind the net, in which
-        // case we snap to either post
-        if (!clamped_goalie_pos)
-        {
-            if (ball.position().y() > 0)
+            // compute angle between two vectors, negative goal post to ball and positive
+            // goal post to ball
+            Angle block_cone_angle =
+                (ball.position() - field.friendlyGoalpostNeg())
+                    .orientation()
+                    .minDiff(
+                        (ball.position() - field.friendlyGoalpostPos()).orientation());
+
+            // compute block cone position, allowing 1 ROBOT_MAX_RADIUS_METERS extra on
+            // either side
+            Point goalie_pos = calcBlockCone(
+                field.friendlyGoalpostNeg(), field.friendlyGoalpostPos(), ball.position(),
+                block_cone_radius * block_cone_angle.toRadians());
+
+            // we want to restrict the block cone to the friendly crease, also potentially
+            // scaled by a defense_area_deflation_parameter
+            Rectangle deflated_defense_area = field.friendlyDefenseArea();
+            deflated_defense_area.expand(-defense_area_deflation);
+
+            // restrain the goalie in the deflated defense area, if the goalie cannot be
+            // restrained or if there is no proper intersection, then we safely default to
+            // center of the goal
+            auto clamped_goalie_pos =
+                restrainGoalieInRectangle(goalie_pos, deflated_defense_area);
+
+            // if the goalie could not be restrained in the deflated defense area,
+            // then the ball must be either on a really sharp angle to the net where
+            // its impossible to get a shot, or the ball is behind the net, in which
+            // case we snap to either post
+            if (!clamped_goalie_pos)
```

Comment:
Just curious, what necessitated these tactic changes? Were there cases where the original tactic was insufficient?

Reply:
There was a section of code at the bottom of the tactic that would overwrite actions given from the if else block.

Reply:
Sorry, not quite sure what you mean here. Could you be a bit more specific?

Reply:
The original tactic didn't work because of the code after line 274. It would override ```next_action``` to always be a move action. So, I moved that code into Case 3, the else statement at line 245 because it seemed to do similar operations, the only noticeable difference being the goalie was more restrained to be closer to the net/ in the defense area and better at sharp angles/ behind the net.

Reply:
I do remember this. Was this something you and @akhilveeraghanta talked about in person that one time?